src/libs/ck-libs/ampi/mpio_globals.h

```diff
@@ -0,0 +1,56 @@
+/* shared header between ROMIO and AMPI's implementation for global variables */
+
+#ifndef MPIO_GLOBALS_H
+#define MPIO_GLOBALS_H
+
+struct ADIOI_Fl_node; /* typedef struct ADIOI_Fl_node ADIOI_Flatlist_node; */
+struct ADIOI_Datarep; /* typedef struct ADIOI_Datarep ADIOI_Datarep; */
+struct ADIOI_FileD; /* typedef struct ADIOI_FileD *ADIO_File; */
+/* typedef MPI_Request ADIO_Request; */
+
+struct ADIO_GlobalStruct {
+    struct ADIOI_Fl_node * ADIOI_Flatlist;
+
+    struct ADIOI_Datarep * ADIOI_Datarep_head;
+
+    /* for f2c and c2f conversion */
+    struct ADIOI_FileD ** ADIOI_Ftable;
+    int ADIOI_Ftable_ptr;
+    int ADIOI_Ftable_max;
+    MPI_Request * ADIOI_Reqtable;
+    int ADIOI_Reqtable_ptr;
+    int ADIOI_Reqtable_max;
+#ifndef HAVE_MPI_INFO
+    MPI_Info *MPIR_Infotable;
+    int MPIR_Infotable_ptr, MPIR_Infotable_max;
+#endif
+#if defined(ROMIO_XFS) || defined(ROMIO_LUSTRE) || 1
```

Comment:
Why the "|| 1"?

Reply:
See the `// TODO: Sync CMK_HAS_LUSTREFS with ROMIO_LUSTRE` in ampiimpl.h. I don't have access to a Lustre machine to test this. Better to always include the two integers in the struct than fail sometimes.